tests/test_date.py

```diff
@@ -118,163 +192,230 @@ def test_should_one_date_for_each_month(self, low, high, expected_results):
+        self.then_should_reject_easily_mistaken_dateutil_arguments(periods)
 
     @parameterized.expand([
-        'years',
-        'months',
-        'days',
-        'hours',
-        'minutes',
-        'seconds',
-        'microseconds',
-        'some_period',
+        param(low=datetime(2014, 4, 15), high=datetime(2014, 4, 14), period='month'),
+        param(low=datetime(2014, 4, 25), high=datetime(2014, 4, 25), period='month'),
     ])
-    def test_should_reject_easily_mistaken_dateutil_arguments(self, invalid_period):
-        low = datetime(2014, 6, 15)
-        period = datetime(2014, 6, 25)
+    def test_empty_period(self, low, high, period):
+        self.given_period_low(low)
+        self.given_period_high(high)
+        self.given_period(period)
+        self.when_intersecting_period_calculated()
+        self.then_period_is_empty()
 
-        with self.assertRaisesRegexp(ValueError, "Invalid period: {}".format(invalid_period)):
-            date.get_intersecting_periods(low, period, period=invalid_period).next()
+    def given_expected_results(self, expected_results):
+        self.expected_results = expected_results
 
-    @parameterized.expand([
-        param(low=datetime(2014, 4, 15), high=datetime(2014, 4, 14)),
-        param(low=datetime(2014, 4, 25), high=datetime(2014, 4, 25)),
-    ])
-    def test_empty_period(self, low, high):
-        results = list(date.get_intersecting_periods(low, high, period='month'))
-        self.assertEquals([], results)
+    def given_dates(self, dates):
+        self.dates = dates
+
+    def given_period_low(self, low):
+        self.period_low = low
+
+    def given_period_high(self, high):
+        self.period_high = high
+
+    def given_period(self, period):
+        self.period = period
+
+    def given_date_parsed(self):
+        self.intersected_dates = self.dates
+
+    def when_intersecting_period_calculated(self):
+        self.intersected_dates = list(date.get_intersecting_periods(
+            self.period_low, self.period_high, period=self.period))
+
+    def then_should_one_date_for_each_month(self):
+        self.assertEquals(self.expected_results, self.intersected_dates)
 
+    def then_date_range_length_is(self, size):
+        self.assertEquals(size, len(self.intersected_dates))
 
-class ParseDateWithFormats(unittest.TestCase):
+    def then_period_low_and_intersected_date_should_equal(self):
+        self.assertEquals(self.period_low, self.intersected_dates[0])
 
-    def test_shouldnt_parse_invalid_date(self):
-        self.assertIsNone(date.parse_with_formats('yesterday', ['%Y-%m-%d'])['date_obj'])
+    def then_period_is_empty(self):
+        self.assertEquals([], self.intersected_dates)
+
+    def when_multiple_periods_are_parsed(self, periods):
+        self.results = []
+        for period in periods:
+            try:
+                result = date.get_intersecting_periods(self.period_low, self.period, period=period).next()
```

Comment:
Please take a moment and try to understand what is happening here. Take a look at other similar methods in this very same file. Tell me if you come up with the solution then.